core/src/main/java/io/questdb/griffin/engine/functions/str/StrPosFunctionFactory.java

```diff
@@ -0,0 +1,113 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.griffin.engine.functions.IntFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+import org.jetbrains.annotations.NotNull;
+
+public class StrPosFunctionFactory implements FunctionFactory {
+
+    @Override
+    public String getSignature() {
+        return "strpos(SS)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        return new Func(args.getQuick(0), args.getQuick(1));
+    }
+
+    public static class Func extends IntFunction implements BinaryFunction {
+
+        private final Function strFunc;
+        private final Function substrFunc;
+
+        public Func(Function strFunc, Function substrFunc) {
+            this.strFunc = strFunc;
+            this.substrFunc = substrFunc;
+        }
+
+        @Override
+        public int getInt(Record rec) {
+            final CharSequence str = this.strFunc.getStr(rec);
+            final CharSequence substr = this.substrFunc.getStr(rec);
+            if (str != null && substr != null) {
+                return strpos(substr, str);
+            }
+            return Numbers.INT_NaN;
+        }
+
+        private int strpos(@NotNull CharSequence substr, @NotNull CharSequence str) {
+            final int strLen = str.length();
+            if (strLen < 1) {
+                return 0;
+            }
+            final int substrLen = substr.length();
+            if (substrLen < 1) {
+                return 1;
+            }
+
+            OUTER:
+            for (int i = 0; i < strLen - substrLen + 1; i++) {
+                final char c = str.charAt(i);
+                if (c == substr.charAt(0)) {
+                    for (int k = 1; k < substrLen && k + i < strLen; k++) {
```

Comment:
I think you're good without `&&` in here

Reply:
Yes, you're right. Remove the `k + i < strLen` check in c9f5cbf9d4ecdbe85365d43feb8404eb999d8805 since `i < strLen - substrLen + 1` invariant is preserved by the outer loop.